openlibrary/templates/books/custom_carousel.html

```diff
@@ -95,9 +33,16 @@ <h2 class="home-h2">
             'analyticsCategory': 'BookCarousel',
             'carouselKey': key,
             'loadMore': {
-              "url": load_more["url"].replace("&amp;", "&"),
+              "queryType": load_more.get("queryType", ""),
+              "q": load_more.get("q", ""),
               "pageMode": load_more.get("mode", "offset"),
-              "limit": load_more.get("limit", 18)
+              "limit": load_more.get("limit", 18),
+              "layout": layout,
+              "key": key,
+              "secondaryAction": secondary_action,
+              "sorts": load_more.get("sorts", ""),
+              "hasFulltextOnly": load_more.get("hasFulltextOnly", True),
+              "subject": load_more.get("subject", "")
```

Comment:
What's the relationship between `subject` and `query` here? Is it an either-or kind of thing where we have some carousels that use the subject field?

If so, I wonder if it might make most sense to serialize the `subject` to the query string so that every search has the same predictable form (i.e. the API only really needs the `q` parameter and not any special cases? Interested in your take.

Reply:
It's an either/or thing.  This is used specifically for `/browse.json` queries, and is eventually passed to the [`compose_ia_url`](https://github.com/internetarchive/openlibrary/blob/f2cc5027d425257dbfa10421a23b9a1caf858448/openlibrary/core/lending.py#L109) function.

`subject` is only used by `custom_ia_carousel`s.  Are we moving towards replacing all of these, or just those that were already replaced by #10532?

Somewhat reated: the `secondaryAction` should be passed to the carousel card template calls. This is passed along with the `/partials` call, but not used and likely important.

Reply:
![Untitled Diagram drawio(1)](https://github.com/user-attachments/assets/2ee5ce51-a507-4d47-b13d-b9f058e602ae)

Not sure if this is helpful or not.  All carousel calls eventually render a `custom_carousel`.